2 dz/4/main.cpp

```diff
@@ -27,11 +27,12 @@ void val(int denominator)
     int firstIndex = 0;
     index -= 1;
     quicksort(fractionValues, firstIndex, index);
+    double epsilon = 0.001;
     for (int i = 0; i <= index; i++)
     {
         for (int j = 0; j <= index; j++)
         {
-            if (fractions[j].arrValue == fractionValues[i])
+            if (abs(fractions[j].arrValue - fractionValues[i]) < epsilon)
```

Comment:
Wow, take it easy :) Yes, we can compare doubles in such a way, but do we need to do it. We can compare nums and dens.